scripts/Imaging/imgpy/filters/crop_coords.py

```diff
@@ -0,0 +1,157 @@
+
+
+def execute(sample, region_of_interest, flat, dark):
+    """
+    Execute the Circular Mask filter.
+
+    :param data: The sample image data as a 3D numpy.ndarray
+    :param region_of_interest: The region of interest that will be cropped
+    :param flat: The average flat image to be cropped
+    :param dark: The average dark image to be cropped
+
+    :return: the data after being processed with the filter
+    """
+    if sample is not None and flat is None and dark is not None:
+        return execute_volume(sample, region_of_interest), None, None
+    else:
+        return execute_volume(sample, region_of_interest), \
+               execute_image(flat, region_of_interest), \
+               execute_image(dark, region_of_interest)
+
+
+def _crop_coords_sanity_checks(coords, data_image, expected_data_shape=3):
+    # if nothing is provided make the user aware
+    if not isinstance(coords, list) or 4 != len(coords):
+        raise ValueError(
+            "Error in crop (region of interest) parameter (expecting a list with four integers) "
+            "[left, top, right, bottom]. Got: {0} with {1}".format(
+                coords, type(coords)))
+
+    elif not all(isinstance(crd, int) for crd in coords):
+        raise ValueError(
+            "Cannot use non-integer coordinates to crop images. Got "
+            "these coordinates: {0}".format(coords))
+
+    elif not isinstance(data_image, np.ndarray) or expected_data_shape != len(
+            data_image.shape):
+        raise ValueError(
+            "Wrong data volume when trying to crop: {0}".format(data_image))
+
+
+def execute_image(data, region_of_interest):
+    """
+
+    :param data :: image as a 2d numpy array
+    :param region_of_interest: coordinates which will be cropped and returned for further processing
+
+    :returns :: cropped data (stack of images)
+
+    """
+
+    if region_of_interest:
+        h.pstart("Starting image cropping with coordinates: {0}. ...".format(
+            region_of_interest))
+
+        data = _crop_image(data, region_of_interest)
+
+        h.pstop(
+            "Finished image cropping with pixel data type: {0}, resulting shape: {1}.".
+            format(data.dtype, data.shape))
+
+    return data
+
+
+def _crop_image(data_image, coords):
+    """
+    Crops a data volume by a rectangle defined by two corner
+    coordinates. Crops along the z axis (outermost numpy array index)
+
+    :param data_image :: image data to be cropped
+    :param coords :: coordinates of the corners that define a rectangle box (crop to this
+    box, as when cropping to the regions of interest).
+    :returns :: cropped image
+    """
+    _crop_coords_sanity_checks(coords, data_image, expected_data_shape=2)
+    left = coords[0]
+    top = coords[1]
+    right = coords[2]
+    bottom = coords[3]
+
+    if not any(coords) or top > bottom or left > right:
+        # skip if for example: 0, 0, 0, 0 (empty selection)
```

Comment:
I rewrote the whole filter, plus the test. It doesn't do any handholding now, if you specify wrong coordinates you will get 0 shaped array out. 
For the future, any restrictions will be enforced by the GUI, for example, right now the rectangle selection will always give you the coordinates in the correct order, so we remove the possibility for the user to make a mistake through that.